tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+        df.columns = ['metric', 'timestamp', 'value']
+        df_features = extract_features(df, column_id='metric', column_sort='timestamp', column_kind=None, column_value=None)
+
+        df_created = None
+        df_assert = None
+        # Test the DataFrame
+        try:
+            df_created = str(df_features.head())
+            if df_created:
+                df_assert = True
+                self.assertEqual(df_assert, True)
+        # Catch when df_created is None
+        except AttributeError:
+            self.assertEqual(df_created, True)
+            pass
+        # Catch if not defined
+        except NameError:
+            self.assertEqual(df_created, True)
+            pass
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Write to disk
+        df_features.to_csv(fname_out)
+
+        if os.path.isfile(fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Transpose
+        df_t = None
+        df_t = df_features.transpose()
+        # df_t.sort_values(by=[0], ascending=[False])
+
+        # Test the DataFrame
+        df_t_created = None
+        df_t_assert = None
+        try:
+            df_t_created = str(df_t.head())
+            if df_t_created:
+                df_t_assert = True
+                self.assertEqual(df_t_assert, True)
+        # Catch when df_t_created is None
+        except AttributeError:
+            self.assertEqual(df_t_created, True)
+            pass
+        # Catch if not defined
+        except NameError:
+            self.assertEqual(df_t_created, True)
+            pass
+
+        # Write the transposed csv
+        df_t.to_csv(t_fname_out)
+
+        if os.path.isfile(t_fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+        return True
+
+    def test_tsfresh_features(self):
+        """
+        This test compares the calculated feature names in the transposed csv to
+        the feature names in the baseline transposed csv and will display new
+        or missing feature names if it is fails.
+        """
+        start = self.test_tsfresh_baseline()
+
+        feature_names_determined = False
+        feature_names = []
+        if start:
+            count_id = 0
+            with open(t_fname_out, 'rt') as fr:
+                reader = csv.reader(fr, delimiter=',')
+                for i, line in enumerate(reader):
+                    if str(line[0]) != '':
+                        if ',' in line[0]:
+                            feature_name = '"%s"' % str(line[0])
+                        else:
+                            feature_name = str(line[0])
+                        count_id += 1
+                        feature_names.append([count_id, feature_name])
```

Comment:
Port logic.  feature names in my app are considered immutable and have a numeric id assigned.  So here the id and names are being checked.